src/openrct2-ui/windows/Ride.cpp

```diff
@@ -2929,13 +2928,15 @@ static void WindowRideVehicleScrollpaint(rct_window* w, rct_drawpixelinfo* dpi,
     startY += rideVehicleEntry->tab_height;
 
     // For each train
+    static VehicleDrawInfo _sprites_to_draw[OpenRCT2::Limits::MaxCarsPerTrain];
```

Comment:
Can you change the name to trainCarImages. This should really be a fixed size vector but it will require refactoring to use. Also move the definition to l2934 and drop the static we don't want to waste time checking if the array is constructed every loop and that should help the optimiser